### PR TITLE
bitbox02: implement sign_message()

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by instagibbs** [comment]
will test

**Comment by benma** [comment]
@instagibbs I also updated the bb02 dep version in a new commit, to enable antiklepto in message signing.

**Review by instagibbs — COMMENTED** [review]
Code looks right to me, output looks right.

Do we want to add notes anywhere to say what software other than Core will validate this since it's not p2pkh?

**Comment by achow101** [comment]
Perhaps we need to have a discussion about what addresses and signature formats are supported for message signing.

Currently Core only supports legacy addresses for address signing and verification. However I know that there are unofficial standards for segwit address signing. So it might be a problem if we support message signing that is incompatible with core.

File: hwilib/devices/bitbox02.py [hunk 1]
```diff
@@ -782,10 +783,32 @@ def script_config_from_utxo(
...
+            message = message.encode("utf-8")
+        keypath = parse_path(bip32_path)
+        purpose = keypath[0]
+        simple_type = {
+            PURPOSE_P2WPKH: bitbox02.btc.BTCScriptConfig.P2WPKH,
+            PURPOSE_P2WPKH_P2SH: bitbox02.btc.BTCScriptConfig.P2WPKH_P2SH,
+        }.get(purpose)
+        if simple_type is None:
+            raise UnavailableActionError(
+                "The keypath bip44 purpose must be 84' or 49'"
```
[the remark this answers is not in the document]
Comment (benma) [reply hwilib/devices/bitbox02.py]: Done

**Comment by benma** [comment]
> Perhaps we need to have a discussion about what addresses and signature formats are supported for message signing.
> 
> Currently Core only supports legacy addresses for address signing and verification. However I know that there are unofficial standards for segwit address signing. So it might be a problem if we support message signing that is incompatible with core.

This PR uses the old 'Bitcoin signed message' format, for all supported addresses. Afaik e.g. the Trezor implementation in HWI does the same, and so does Electrum. I think Trezor has an extension that signs segwit a bit differently, but it does not seem to be in use in third party integrations like HWI and Electrum. I am not aware of any other unofficial formats.

I am not sure about what Core does with segwit addresses, but the wider ecosystem (Electrum, various brokers that require signed msgs, etc.) accept the same format regardless of whether it is legacy or segwit.

Maybe BIP322 is an avenue, but for the time being, this PR is compatible with how other hw wallet vendors and wallets do it.

**Comment by benma** [comment]
@achow101 ready to merge? As mentioned, this PR behaves the same as e.g. Trezor.

**Comment by achow101** [comment]
ACK a30d672e56ee142d68a399f8ecd95af7d809e287

**Pull request merged**